### ISSUE TITLE
Understanding logic

### ISSUE THREAD

**Issue by keerthana1512** [artitw/text2text#2]
Hey..
This project is exactly what I was searching for. I have tried many projects on github for question generation but found none satisfactory. This has helped me a lot. I am trying to understand the logic behind question generation from text. I have understood that you are using bert-model for generation. Can you please help me in understanding this better?

**Comment by artitw** [comment]
Yes, you are on the right track. The backend model is similar to BERT with modifications to perform text generation. If you find this project useful, please share back your experiences and feedback with the community.

**Comment by secsrexion** [comment]
Hello 
I hope you are doing fine , firstly i thank you for your contributions on question generation , and i have a question if i may ask .
Im trying to build a question generation system for a non-English language i was planing to use UniLm ( miniLm multilingual version ) because bert is not really built for text generation since you have experience on that what how do you suggest to do that and am i following the good path .

Thank you in advance for your appreciated help !

**Comment by madhoolika98** [comment]
Hey 
I just wanted to ask you if you have followed any paper or if you could give me any source link to which you have followed regarding the question framing of bert model. I just want to understand how  exactly bert is choosing a particular wh question for a particular sentence. and how the question processing is happening. it would be really helpful if u can help me with that

**Comment by artitw** [comment]
In simplified terms, the question generation was trained by augmenting BERT to perform sequence-to-sequence [1] using inverted SQuAD [2]. 

[1] https://arxiv.org/abs/1905.03197
[2] https://arxiv.org/abs/1909.05017
